source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -783,6 +795,13 @@ connection, for example:
         serverAddress = serverAddress
         connectTimeoutMS = connectTimeoutMS
         heartbeatFrequencyMS = heartbeatFrequencyMS
+
+        # Heartbeat command
+        if versionedApi or handshakeResponse.helloOk:
```

Comment:
Similar comment here about `handshakeResponse`. Since the `RttMonitor` is independent. To go along with the suggestion about storing a `ServerDescription.helloOk`, the `RttMonitor` may need to start storing a `ServerDescription` to track whether it can use `hello`, and reset that on an exception in `run`.

Reply:
Thinking this through a bit, I believe we should leave these threads independent from each other. My suggestion is to initialize the `RttMonitor` with `helloOk` based on `apiVersion` - same as the main monitoring thread. The `RttMonitor` can then examine the response to determine if subsequent pings should use hello or legacy hello. This way the `RttMonitor` does not need access to the `ServerDescription` and we don't have to worry about synchronization between the main monitor thread attempting to update the `RttMonitor` with the current state of `helloOk`.

Reply:
Good thinking.